### ISSUE TITLE
Add support for loading telegraf classes from file instead of K8S-secret

### ISSUE THREAD

**Issue by gitirabassi** [influxdata/telegraf-operator#20]
A kubernetes secret can be mounted as file also

This will help us in moving all secrets to be managed by Hashicorp Vault. 

Using vault-agent we can have the classes on disk directly from a secret in Vault

**Comment by wojciechka** [comment]
@gitirabassi can you describe how the files will be mapped, especially in the context of vault?

I assume it will be that each class is a file. Then the logic is to add `--telegraf-classes-directory=/vault` and for class name "app", the operator should just read `/vault/app` file ?

**Comment by gitirabassi** [comment]
I think it would be better not to repeat the base path selecting the default class. 
since they are files I think it's better having them with an extention by defualt, but it can work even without one


```
--telegraf-classes-directory=/vault/secrets
--telegraf-default-class=infra.conf
```

in which `infra.conf` needs to be present in the folder `/vault/secrets`

PS. `/vault/secrets` is the default folder used by the `vault-agent` to place secrets

**Comment by gitirabassi** [comment]
I think this issue is very tied together with with https://github.com/influxdata/telegraf-operator/issues/19

It could even be a single PR for both issues

**Comment by rawkode** [comment]
Is it possible to store Telegraf configs as Config Maps and interpolate the secrets at runtime?

**Comment by wojciechka** [comment]
> Is it possible to store Telegraf configs as Config Maps and interpolate the secrets at runtime?

After some talking with @gitirabassi, we've decided to simplify the logic and only use filesystem for reading the classes. This is what I did in #24.

This allows using a secret (and `deploy/dev.yml` will do just that). But a config map or using any directory - such as using vault sidecar - can also be used.

I am curious about your specific use case. What would that solve? 

Contents of classes can be config map and then refer to environment variables that can be added using `envFrom` in the container definition.

I think moving to a directory-based configuration makes it more convenient and more powerful.

But if you had another use case in mind, then let me know. We can talk more about how to either enable it in #24 or how to use changes there to do it.
